### PR TITLE
CSP reporting integration

### PR DESCRIPTION
**Pull request by hestonhoffman**
See [DOCS-3551](https://datadoghq.atlassian.net/browse/DOCS-3551)

Ready to merge

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

[DOCS-3551]: https://datadoghq.atlassian.net/browse/DOCS-3551?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by aviat — APPROVED** [review]
This looks good to me, I think we can proceed!

**Comment by hestonhoffman** [comment]
I need to get a logo for the integrations page before I publish this, so I'll work on that

File: content/en/integrations/content_security_policy_logs.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
+---
+categories:
+- log collection
+- security
+description: 'Detect and aggregate CSP violations with Datadog'
+doc_link: /integrations/content_security_policy_logs/
+dependencies:
+    ['https://github.com/DataDog/documentation/blob/master/content/en/integrations/content_security_policy_logs.md']
+has_logo: true
+integration_title: Content Security Policy logs
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:10]:
```suggestion
integration_title: Content Security Policy Logs
```

File: content/en/integrations/content_security_policy_logs.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+name: content_security_policy_logs
+public_title: Datadog-Content Security Policy logs
+short_description: 'Detect CSP violations'
+version: '1.0'
+integration_id: "content_security_policy_logs"
+---
+
+## Overview
+
+The Datadog Content Security Policy (CSP) integration sends logs to Datadog from web browsers as they interpret your CSP and detect violations. Using the CSP integration means you don't have to host or manage a dedicated endpoint to aggregate your CSP data.
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:22]:
```suggestion
The Datadog Content Security Policy (CSP) integration sends logs to Datadog from web browsers as they interpret your CSP and detect violations. By using the CSP integration, you don't have to host or manage a dedicated endpoint to aggregate your CSP data.
```

File: content/en/integrations/content_security_policy_logs.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+For more information about CSPs, see [Google's web.dev post][1].
+
+## Prerequisites
+
+Before you add a directive to a CSP header, [generate a client token in your Datadog account][2].
+
+<div class="alert alert-info">API keys cannot be used to configure the browser logs SDK, because the keys would be exposed client-side. To collect logs from web browsers, use a client token. <a href="https://docs.datadoghq.com/logs/log_collection/?tab=host#setup">See the client token documentation for more details</a>.</div>
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:30]:
```suggestion
<div class="alert alert-info">For security reasons, you must use a client token to collect logs from web browsers. You cannot use Datadog API keys to configure the Datadog Browser Logs SDK as they would be exposed client-side. For more information, see the <a href="https://docs.datadoghq.com/logs/log_collection/?tab=host#setup">client token documentation for more details</a>.</div>
```

File: content/en/integrations/content_security_policy_logs.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+You need a URL where browsers can send policy violation reports. The URL must have the following format:
+
+```
+https://csp-report.browser-intake-datadoghq.com/api/v2/logs?dd-api-key=<client -token>&dd-evp-origin=content-security-policy&ddsource=csp-report
+```
+
+Optionally, add the `ddtags` key to the URL to convey additional information such as the service name, the environment, and service version:
+- `env`: the application's environment
+- `service`: the service name for your application
+- `version`: the application's version
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:43]:
```suggestion
- `env`: the application's environment.
- `service`: the service name for your application.
- `version`: the application's version.
```

File: content/en/integrations/content_security_policy_logs.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    'blocked-uri': 'https://evil.com',
+    'status-code': 200
+  }
+}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+[1]: https://web.dev/csp/
+[2]: https://app.datadoghq.com/organization-settings/client-tokens
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:148]:
```suggestion
[2]: https://app.datadoghq.com/organization-settings/client-tokens
[3]: /getting_started/tagging/unified_service_tagging
```

File: content/en/integrations/content_security_policy_logs.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+## Prepare a URL for the CSP
+
+You need a URL where browsers can send policy violation reports. The URL must have the following format:
+
+```
+https://csp-report.browser-intake-datadoghq.com/api/v2/logs?dd-api-key=<client -token>&dd-evp-origin=content-security-policy&ddsource=csp-report
+```
+
+Optionally, add the `ddtags` key to the URL to convey additional information such as the service name, the environment, and service version:
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:40]:
```suggestion
Optionally, add the `ddtags` key (service name, the environment, and service version) to the URL to set up [Unified Service Tagging][3]:
```

File: content/en/integrations/content_security_policy_logs.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+service%3AbillingService%2Cenv%3Aproduction
+```
+
+And the final URL with tags would be:
+
+```
+https://csp-report.browser-intake-datadoghq.com/api/v2/logs?dd-api-key=<client -token>&dd-evp-origin=content-security-policy&ddsource=csp-report&ddtags=service%3AbillingService%2Cenv%3Aproduction
+```
+
+## Adding the URL to the CSP
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:62]:
```suggestion
## Add the URL to the CSP
```

File: content/en/integrations/content_security_policy_logs.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+
+You can either embed the URL in an HTTP header (recommended), or embed it in a `<meta>` HTML tag.
+
+### Embed the policy in an HTTP header
+
+This is the recommended way to embed a Content Security Policy. You can either use the `report-uri` directive or the `report-to` directive. The `report-to` directive will eventually supersede `report-uri`, but is not yet supported by all browsers.
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:68]: Not sure about the temporal element of the second sentence here:

```suggestion
Datadog recommends embedding the Content Security Policy in an HTTP header. You can either use the `report-uri` directive or the `report-to` directive. The `report-to` directive will eventually supersede `report-uri`, but is not yet supported by all browsers.
```

File: content/en/integrations/content_security_policy_logs.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  ```json
+  Content-Security-Policy: ...; report-to browser-intake-datadoghq
+  Report-To: { "group": "browser-intake-datadoghq",
+              "max_age": 10886400,
+              "endpoints": [
+                  { "url": " https://csp-report.browser-intake-datadoghq.com/api/v2/logs?dd-api-key=<client -token>&dd-evp-origin=content-security-policy&ddsource=csp-report" }
+              ] }
+  ```
+
+### Policy embedded in a <meta> HTML tag
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:85]: The preview is skipping the `meta` part.

```suggestion
### Policy embedded in a `<meta>` HTML tag
```

File: content/en/integrations/content_security_policy_logs.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+You can also embed the URL in an HTML <meta> tag.
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:87]: The preview is skipping the `meta` part.

```suggestion
You can also embed the URL in a `<meta>` HTML tag.
```

File: content/en/integrations/content_security_policy_logs.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+    ['https://github.com/DataDog/documentation/blob/master/content/en/integrations/content_security_policy_logs.md']
+has_logo: true
+integration_title: Content Security Policy logs
+is_public: true
+kind: integration
+name: content_security_policy_logs
+public_title: Datadog-Content Security Policy logs
+short_description: 'Detect CSP violations'
+version: '1.0'
+integration_id: "content_security_policy_logs"
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:17]:
```suggestion
integration_id: "content_security_policy_logs"
further_reading:
  - link: "/getting_started/tagging/unified_service_tagging/"
    tag: "Documentation"
    text: "Learn about Unified Service Tagging"
```

File: content/en/integrations/content_security_policy_logs.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+    'effective-directive': 'script-src',
+    'original-policy': 'trusted-types toto; script-src good.com; report-uri http://127.0.0.1:8000/csp_reports',
+    'blocked-uri': 'https://evil.com',
+    'status-code': 200
+  }
+}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
```
Comment (alai97) [inline content/en/integrations/content_security_policy_logs.md:146]:
```suggestion

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

```

**Review by alai97 — CHANGES_REQUESTED** [review]
Doc review~

**Review by alai97 — APPROVED** [review]
Looks good!

**Review by alai97 — APPROVED** [review]
Looks good!

**Pull request merged**